### PR TITLE
Add volunteer type for NFC/event scanning

### PR DESCRIPTION
**Pull request by cktang88**
The way this is meant to work is that volunteers who are scanning NFC will be assigned a `usertype.volunteer`

### CODE REVIEW

File: src/client/assets/routes.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ const routes = [
 		path: '/help',
 	},
 	{
-		authLevel: [UserType.Organizer, UserType.Sponsor],
+		authLevel: [UserType.Organizer, UserType.Volunteer, UserType.Sponsor],
```
Comment (slowjazz) [inline src/client/assets/routes.js]: Sponsors are checking people in?

Comment (cktang88) [reply src/client/assets/routes.js]: at their booth

File: src/server/resolvers/index.ts [hunk 2]
```diff
@@ -164,7 +164,11 @@ export const resolvers: CustomResolvers<Context> = {
 			return addOrUpdateEvent(input, models);
 		},
 		checkInUserToEvent: async (root, { input }, { models, user }) => {
-			checkIsAuthorized(UserType.Organizer, user);
+			try {
+				checkIsAuthorized(UserType.Organizer, user);
+			} catch (e) {
+				checkIsAuthorized(UserType.Volunteer, user);
+			}
```
Comment (slowjazz) [inline src/server/resolvers/index.ts]: Add a case for Sponsors

File: src/server/resolvers/index.ts [hunk 3]
```diff
@@ -265,7 +269,11 @@ export const resolvers: CustomResolvers<Context> = {
 			return companyCreated;
 		},
 		checkInUserToEventByNfc: async (root, { input }, { models, user }) => {
-			checkIsAuthorized(UserType.Organizer, user);
+			try {
+				checkIsAuthorized(UserType.Organizer, user);
+			} catch (e) {
+				checkIsAuthorized(UserType.Volunteer, user);
+			}
```
Comment (slowjazz) [inline src/server/resolvers/index.ts]: Same - add case for sponsors

File: src/server/resolvers/index.ts [hunk 4]
```diff
@@ -359,17 +367,29 @@ export const resolvers: CustomResolvers<Context> = {
 			return ret;
 		},
 		registerNFCUIDWithUser: async (root, { input }, { models, user }) => {
-			checkIsAuthorized(UserType.Organizer, user);
+			try {
+				checkIsAuthorized(UserType.Organizer, user);
+			} catch (e) {
+				checkIsAuthorized(UserType.Volunteer, user);
+			}
```
Comment (slowjazz) [inline src/server/resolvers/index.ts]: Case for sponsors

File: src/server/resolvers/index.ts [hunk 5]
```diff
@@ -359,17 +367,29 @@ export const resolvers: CustomResolvers<Context> = {
...
 			const userRet = await registerNFCUIDWithUser(input.nfcid, input.user, models);
 			return userRet;
 		},
 		removeUserFromEvent: async (root, { input }, { models, user }) => {
-			checkIsAuthorized(UserType.Organizer, user);
+			try {
+				checkIsAuthorized(UserType.Organizer, user);
+			} catch (e) {
+				checkIsAuthorized(UserType.Volunteer, user);
+			}
```
Comment (slowjazz) [inline src/server/resolvers/index.ts]: Case for sponsors

File: src/server/resolvers/index.ts [hunk 6]
```diff
@@ -359,17 +367,29 @@ export const resolvers: CustomResolvers<Context> = {
...
 			const userRet = await removeUserFromEvent(input.user, input.event, models);
 			return userRet;
 		},
 		removeUserFromEventByNfc: async (root, { input }, { models, user }) => {
-			checkIsAuthorized(UserType.Organizer, user);
+			try {
+				checkIsAuthorized(UserType.Organizer, user);
+			} catch (e) {
+				checkIsAuthorized(UserType.Volunteer, user);
+			}
```
Comment (slowjazz) [inline src/server/resolvers/index.ts]: Case for sponsors

File: src/server/resolvers/index.ts (hunk 2 above)
Comment (slowjazz) [reply src/server/resolvers/index.ts]: You can also modify `checkIsAuthorized` to take an array of UserTypes

**Comment by cktang88** [comment]
decided not to modify original func b/c that's used in a many places

**Comment by cktang88** [comment]
it's really sad that this PR doesn't work atm

**Pull request merged**